.evergreen/github_app/create_or_modify_comment.mjs

```diff
@@ -0,0 +1,96 @@
+/**
+ * Create or modify a GitHub comment using the mongodb-drivers-comment-bot.
+ */
+import * as fs from "fs";
+import * as process from "process";
+import { program } from 'commander';
+import { App } from "octokit";
+
+const appId = process.env['GITHUB_APP_ID'];
+const privateKey = process.env['GITHUB_SECRET_KEY'].replace(/\\n/g, '\n');
+if (appId == '' || privateKey == '') {
+    console.error("Missing GitHub App auth information");
+    process.exit(1)
+}
+
+// Handle cli.
+program
+  .version('1.0.0', '-v, --version')
+  .usage('[OPTIONS]...')
+  .requiredOption('-o, --repo-owner <owner>', 'The owner of the repo (e.g. mongodb).')
+  .requiredOption('-n, --repo-name <name>', 'The name of the repo (e.g. mongo-go-driver).')
+  .requiredOption('-m, --body-match <string>', 'The comment body to match')
+  .requiredOption('-c, --comment-path <path>', 'The path to the comment body file')
+  .requiredOption('-h, --head-sha <sha>', 'The sha of the head commit')
+  .parse(process.argv);
+
+const options = program.opts();
+const owner = options.repoOwner;
+const repo = options.repoName;
+const bodyMatch = options.bodyMatch;
+const bodyText = fs.readFileSync(options.commentPath, { encoding: 'utf8' });
+const targetSha = options.headSha;
+
+// Set up the app.
+const installId = process.env['GITHUB_APP_INSTALL_ID_' + owner.toUpperCase()];
+if (installId == '') {
+    console.error(`Missing install id for ${owner}`)
+    process.exit(1)
+}
+const app = new App({ appId, privateKey });
+const octokit = await app.getInstallationOctokit(installId);
+const headers =  {
+    "x-github-api-version": "2022-11-28",
+};
+
+// Find the matching pull request.
+let issueNumber = -1;
+let resp = await octokit.request("GET /repos/{owner}/{repo}/pulls?state=open&per_page=100", {
+    owner,
+    repo,
+    headers
+});
+for (const pull of resp['data']) {
+    if (pull['head']['sha'] == targetSha) {
+        issueNumber = pull['number'];
+    }
+}
+if (issueNumber == -1) {
+    console.error(`ERROR: Could not find matching pull request for sha ${targetSha}`)
+    process.exit(1)
+}
+
+// Find a matching comment if it exists, and update it.
+var found = false
+resp = await octokit.request("GET /repos/{owner}/{repo}/issues/{issue_number}/comments", {
+    owner,
+    repo,
+    issue_number: issueNumber,
+    headers
+});
+resp.data.forEach(async (comment) => {
+    if (comment.body.includes(bodyMatch)) {
+        if (found) {
+            return
+        }
+        found = true;
+        await octokit.request("PATCH /repos/{owner}/{repo}/issues/comments/{comment_id}", {
+            owner,
+            repo,
+            body: bodyText,
+            comment_id: comment.id,
+            headers
+        });
+    }
+})
```

Comment:
In Node, you need to be careful of using `async` functions as callbacks where they're not expected, because if the async function throws an error, it won't be caught and the program crashes.  Here I'd either use Array.find() again, or use a for-of loop 
```suggestion
const comment = resp.data.find(comment => comment.body.includes(bodyMatch));
if (!comment) {
  // create comment
} 

// update comment
```

or

```javascript
for (const comment of resp.data)
    if (comment.body.includes(bodyMatch)) {
        found = true;
        await octokit.request("PATCH /repos/{owner}/{repo}/issues/comments/{comment_id}", {
            owner,
            repo,
            body: bodyText,
            comment_id: comment.id,
            headers
        });
        break;
    }
}
```

Reply:
👍🏼